### PR TITLE
Creation of request and interface method for Workflow raise event

### PR DESCRIPTION
**Pull request by RyanLettieri**
# Description

Creation of raise event method and request for workflow.

Note that this PR needs to close before https://github.com/dapr/dapr/pull/6049 can complete

## Issue reference

This PR will close: #[_[6045]_](https://github.com/dapr/dapr/issues/6045)

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: workflows/requests.go
Comment (cgillum) [inline workflows/requests.go:19]: Do we get any benefit from using `interface{}` here? I wonder if it would be easier to simply use `[]byte` since it's always going to be some opaque serialized payload anyways?

**Review by cgillum — DISMISSED** [review]
LGTM!

File: workflows/requests.go
Comment (artursouza) [inline workflows/requests.go:17]: Should this be WorkflowReference instead to be future proof?

File: workflows/requests.go
Comment (artursouza) [inline workflows/requests.go:9]: Can we remove WorkflowReference then?

**Review by artursouza — CHANGES_REQUESTED** [review]

File: workflows/requests.go
Comment (artursouza) [reply workflows/requests.go:9]: Please, follow up on this one and if it can be removed, cut a new PR.

**Review by artursouza — APPROVED** [review]

**Review by ItalyPaleAle — APPROVED** [review]

File: workflows/requests.go
Comment (RyanLettieri) [reply workflows/requests.go:9]: Here's the issue to track this suggestion: https://github.com/dapr/dapr/issues/5911

**Comment by ItalyPaleAle** [comment]
Approved, but please don't merge until we know the PR in the runtime is approved. We'll then need to do something about the cert tests failing.

**Comment by ItalyPaleAle** [comment]
Because dapr/dapr#6049 is no close to being ready to be merged, and with this PR merged in master our cert tests are broken (and so is our ability to merge any PR in dapr/components-contrib), we are going to revert this PR.

Once the runtime PR is ready, please re-open a PR to get the changes merged

**Comment by berndverst** [comment]
Agreed. This needs to be reverted for now. It should never have been merged in its current state.

**Pull request merged**
